GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m

```diff
@@ -102,74 +102,69 @@ + (NSString *)libraryDataStoragePath {
-                    qosTier:(nullable NSNumber *)qosTier
-                  mappingID:(nullable NSString *)mappingID {
-  NSString *baseEventPath = [GDTCORFlatFileStorage baseEventStoragePath];
-  // If only a target was given, return the target path.
-  if (qosTier == nil && mappingID == nil) {
-    return [NSString stringWithFormat:@"%@/%ld", baseEventPath, (long)target];
-  }
-
-  // If only a target and mappingID were given, return the mapping ID path.
-  if (qosTier == nil) {
-    return [NSString stringWithFormat:@"%@/%ld/%@", baseEventPath, (long)target, mappingID];
-  }
-
-  // If only a target and qosTier were given, return the QoS tier path.
-  if (mappingID == nil) {
-    return [NSString stringWithFormat:@"%@/%ld/%@", baseEventPath, (long)target, qosTier];
-  }
-
-  // If a target, mappingID, and qosTier were all given, return a single target/qosTier/mappingID
-  // directory.
+                    eventID:(NSNumber *)eventID
+                    qosTier:(NSNumber *)qosTier
+                  mappingID:(NSString *)mappingID {
   return
-      [NSString stringWithFormat:@"%@/%ld/%@/%@", baseEventPath, (long)target, qosTier, mappingID];
+      [NSString stringWithFormat:@"%@/%ld/%@.%@.%@", [GDTCORFlatFileStorage baseEventStoragePath],
+                                 (long)target, eventID, qosTier, mappingID];
 }
 
-+ (NSArray<NSString *> *)searchPathsWithEventSelector:(GDTCORStorageEventSelector *)eventSelector {
-  NSMutableArray<NSString *> *searchPaths = [[NSMutableArray alloc] init];
-  if (eventSelector.selectedQosTiers && eventSelector.selectedQosTiers.count > 0) {
-    for (NSNumber *qosTier in eventSelector.selectedQosTiers) {
-      NSString *searchPath = [self pathForTarget:eventSelector.selectedTarget
-                                         qosTier:qosTier
-                                       mappingID:eventSelector.selectedMappingID];
-      BOOL isDirectory;
-      if ([[NSFileManager defaultManager] fileExistsAtPath:searchPath isDirectory:&isDirectory]) {
-        if (isDirectory) {
-          [searchPaths addObject:searchPath];
-        }
-      }
++ (NSSet<NSString *> *)pathsForTarget:(GDTCORTarget)target
+                             eventIDs:(nullable NSSet<NSNumber *> *)eventIDs
+                             qosTiers:(nullable NSSet<NSNumber *> *)qosTiers
+                           mappingIDs:(nullable NSSet<NSString *> *)mappingIDs {
+  NSMutableSet<NSString *> *paths = [[NSMutableSet alloc] init];
+  NSFileManager *fileManager = [NSFileManager defaultManager];
+  NSString *targetPath = [NSString
+      stringWithFormat:@"%@/%ld", [GDTCORFlatFileStorage baseEventStoragePath], (long)target];
+  NSError *error;
+  NSArray<NSString *> *dirPaths = [fileManager contentsOfDirectoryAtPath:targetPath error:&error];
+  if (error) {
+    GDTCORLogDebug(@"There was an error reading the contents of the target path: %@", error);
+    return paths;
+  }
+  BOOL checkingIDs = eventIDs != nil && eventIDs.count > 0;
```

Comment:
nit: Taking into account Objective C `nil` behaviour  `eventIDs != nil` looks redundant, because `nil` casted to an integer as `0`.

Reply:
doh, yeah, good catch